src/main/drivers/bus_spi_impl.h

```diff
@@ -87,7 +87,7 @@ typedef struct SPIDevice_s {
 extern spiDevice_t spiDevice[SPIDEV_COUNT];
 
 void spiInitDevice(SPIDevice device);
-void spiInternalInitStream(const extDevice_t *dev, bool preInit);
+void spiInternalInitStream(const extDevice_t *dev, volatile busSegment_t *segment);
```

Comment:
this `volatile` is only breaking code optimization. `segment` passed here must not be processed (yet), so there is nothing that can change it asynchronously. Otherwise it would lead to all kinds of nasty race conditions.

IMO the problem is that whole linked list is declared `volatile` even when single segment can be active. Ith thous be handled other way around - cast potentially active segment as volatile, only where it is important.